### ISSUE TITLE
Old (3.4.4) score opens two times longer in 4.2-4.4 (comparing with 4.0-4.1.1)

### ISSUE THREAD

**Issue by DmitryArefiev** [musescore/MuseScore#21620]
### Issue type

Opening/saving issue

### Steps to reproduce

1. Open [Mvt.4_Beethoven__Symphony_No._9__Op._125.zip](https://github.com/musescore/MuseScore/files/14361865/Mvt.4_Beethoven__Symphony_No._9__Op._125.zip) in **4.1.1** and measure time
2. Open [Mvt.4_Beethoven__Symphony_No._9__Op._125.zip](https://github.com/musescore/MuseScore/files/14361865/Mvt.4_Beethoven__Symphony_No._9__Op._125.zip) in **4.2** - **4.4** and measure time

### MuseScore Version

4.3

### Regression

Yes, this used to work in a previous version of MuseScore 4.x

### Operating system

Windows 11

**Comment by mathesoncalum** [comment]
There are a handful of commits that have contributed to the increased load times, though #17983 and #19659 seem to have had the most significant effect.

**Current Master:**
![master](https://github.com/musescore/MuseScore/assets/47119327/c2588f60-d4a5-4f0f-85b6-23f709c1fa35)

**After reverting the changes from these PRs:**
![revert](https://github.com/musescore/MuseScore/assets/47119327/5d5093c4-8042-43f6-abb0-0cde70d8c6e1)

Looking into potential optimizations now.

**Comment by DmitryArefiev** [comment]
This was improved after #21803

The new results added to description (Summary is also updated)

**Comment by bkunda** [comment]
@DmitryArefiev pls check to see if this is still reproducible (also in 4.3.1).

**Comment by DmitryArefiev** [comment]
Yes, the same result: Old (3.4.4) score opens 2 times longer in **4.3.1** and **master**

But I think we can bump it to 4.5 for now. Please change if don't agree
